mantidproject/mantid
# Add TeixeiraWaterIqt to IqtFit

### Description of work
This PR adds the `TeixeiraWaterIqt` fit function to the IqtFit tab of the inelastic QENS Fitting interface. It makes sure that the q values for the fit function are updated when changing the plotted spectrum.

Fixes #36502

### To test:

1. Open Inelastic QENS Fitting interface
2. Go to IqtFit
3. Load the following data
4. Select the `TeixeiraWaterIqt` option in Fit Type
5. It should have three parameters
6. Select "See full function"
7. Check that the Q value updates whenever you change the plotted spectrum
8. Click "Run" and wait
9. It should successfully perform a fit

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
